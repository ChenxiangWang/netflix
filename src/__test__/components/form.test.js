import React from "react";
import { render, screen } from "@testing-library/react";
import Form from "../../components/form";

import { BrowserRouter, Router } from "react-router-dom";

jest.mock("react-router-dom");

// TODO: Jest can not render the <Link/> correctly
describe("<Form/> test", () => {
  it("test renders", () => {
    const { container } = render(
      <Form>
        <Form.Title>Sign In</Form.Title>
        <Form.Base method="POST">
          <Form.Input
            data-testid="input"
            placeholder="Email Address"
            onChange={() => {}}
          />

          <Form.Input
            type="password"
            placeholder="Password"
            autoComplete="off"
            onChange={() => {}}
          />

          <Form.Submit data-testid="submit" disabled type="submit">
            Sign In
          </Form.Submit>
        </Form.Base>

        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not bot.
          Learn nore.
        </Form.TextSmall>
      </Form>
    );
    expect(screen.getByText("New to Netflix?")).toBeTruthy();
    expect(
      screen.getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not bot. Learn nore."
      )
    ).toBeTruthy();
    expect(screen.getByTestId("submit")).toBeTruthy();
    expect(screen.getByTestId("submit").disabled).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
