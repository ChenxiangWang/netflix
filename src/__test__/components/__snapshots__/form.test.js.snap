// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> test test renders 1`] = `
<div
  class="style__Container-sc-1kjg77i-0 cpCtQb"
>
  <h1
    class="style__Title-sc-1kjg77i-3 doDMYM"
  >
    Sign In
  </h1>
  <form
    class="style__Base-sc-1kjg77i-2 dUgePX"
    method="POST"
  >
    <input
      class="style__Input-sc-1kjg77i-7 eDxTAW"
      data-testid="input"
      placeholder="Email Address"
    />
    <input
      autocomplete="off"
      class="style__Input-sc-1kjg77i-7 eDxTAW"
      placeholder="Password"
      type="password"
    />
    <button
      class="style__Submit-sc-1kjg77i-8 iHUjzB"
      data-testid="submit"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="style__Text-sc-1kjg77i-4 cvWRRt"
  >
    New to Netflix? 
  </p>
  <p
    class="style__TextSmall-sc-1kjg77i-5 gtPprl"
  >
    This page is protected by Google reCAPTCHA to ensure you're not bot. Learn nore.
  </p>
</div>
`;
